--- a/nginx.spec	2019-05-30 13:28:20.000000000 +0000
+++ b/nginx.spec	2019-07-25 09:09:20.596537000 +0000
@@ -37,6 +37,20 @@
 Source12:          nginx.conf
 Source13:          nginx-upgrade
 Source14:          nginx-upgrade.8
+Source21:          ngx_image_thumb.tar.gz
+Source22:          nginx-http-rdns.tar.gz
+Source23:          nginx-rtmp-module.tar.gz
+Source24:          ngx_devel_kit.tar.gz
+Source25:          lua-nginx-module.tar.gz
+Source26:          echo-nginx-module.tar.gz
+Source27:          headers-more-nginx-module.tar.gz
+Source28:          redis2-nginx-module.tar.gz
+Source29:          naxsi.tar.gz
+Source30:          naxsi_core.rules
+Source31:          nginx-module-vts.tar.gz
+Source32:          nginx-lua-prometheus.tar.gz
+Source33:          stream-lua-nginx-module.tar.gz
+Source34:          nginx-upsync-module.tar.gz
 Source102:         nginx-logo.png
 Source103:         404.html
 Source104:         50x.html
@@ -50,6 +64,10 @@
 # downstream patch - changing logs permissions to 664 instead
 # previous 644
 Patch2:            nginx-1.12.1-logs-perm.patch
+Patch3:            nginx-client-addr.patch
+# https://github.com/intaro/nginx-image-filter-watermark.git
+# https://github.com/tyler/image_filter_module
+Patch4:            nginx-image-filter.patch
 
 BuildRequires:     gcc
 %if 0%{?with_gperftools}
@@ -67,6 +85,10 @@
 Requires:          nginx-all-modules = %{epoch}:%{version}-%{release}
 %endif
 
+Requires:          luajit
+%if 0%{?with_gperftools}
+Requires:          gperftools
+%endif
 Requires:          openssl
 Requires:          pcre
 Requires(pre):     nginx-filesystem
@@ -170,8 +192,23 @@
 
 %prep
 %setup -q
+%setup -T -D -a 21
+%setup -T -D -a 22
+%setup -T -D -a 23
+%setup -T -D -a 24
+%setup -T -D -a 25
+%setup -T -D -a 26
+%setup -T -D -a 27
+%setup -T -D -a 28
+%setup -T -D -a 29
+%setup -T -D -a 31
+%setup -T -D -a 32
+%setup -T -D -a 33
+%setup -T -D -a 34
 %patch0 -p0
 %patch2 -p1
+%patch3 -p1 -b .orig
+%patch4 -p1 -b .orig
 cp %{SOURCE200} %{SOURCE210} %{SOURCE10} %{SOURCE12} .
 
 %if 0%{?rhel} > 0 && 0%{?rhel} < 8
@@ -241,6 +278,17 @@
     --with-google_perftools_module \
 %endif
     --with-debug \
+    --add-module=%{_builddir}/nginx-%{version}/lua-nginx-module \
+    --add-module=%{_builddir}/nginx-%{version}/echo-nginx-module \
+    --add-module=%{_builddir}/nginx-%{version}/naxsi/naxsi_src \
+    --add-module=%{_builddir}/nginx-%{version}/ngx_devel_kit \
+    --add-module=%{_builddir}/nginx-%{version}/headers-more-nginx-module \
+    --add-module=%{_builddir}/nginx-%{version}/nginx-module-vts \
+    --add-module=%{_builddir}/nginx-%{version}/redis2-nginx-module \
+    --add-module=%{_builddir}/nginx-%{version}/nginx-rtmp-module \
+    --add-module=%{_builddir}/nginx-%{version}/ngx_image_thumb \
+    --add-module=%{_builddir}/nginx-%{version}/nginx-http-rdns \
+    --add-module=%{_builddir}/nginx-%{version}/nginx-upsync-module \
     --with-cc-opt="%{optflags} $(pcre-config --cflags)" \
     --with-ld-opt="$nginx_ldopts"; then
   : configure failed
@@ -281,6 +329,15 @@
 install -p -m 0644 ./nginx.conf \
     %{buildroot}%{_sysconfdir}/nginx
 
+install -p -m 0644 %{SOURCE30} \
+    %{buildroot}%{_sysconfdir}/nginx
+
+install -p -m 0644 %{_builddir}/nginx-%{version}/nginx-lua-prometheus/prometheus.lua \
+    %{buildroot}%{_sysconfdir}/nginx
+
+install -p -m 0644 %{_builddir}/nginx-%{version}/nginx-lua-prometheus/prometheus_test.lua \
+    %{buildroot}%{_sysconfdir}/nginx
+
 rm -f %{buildroot}%{_datadir}/nginx/html/index.html
 ln -s ../../fedora-testpage/index.html \
       %{buildroot}%{_datadir}/nginx/html/index.html
@@ -406,6 +463,9 @@
 %config(noreplace) %{_sysconfdir}/nginx/mime.types.default
 %config(noreplace) %{_sysconfdir}/nginx/nginx.conf
 %config(noreplace) %{_sysconfdir}/nginx/nginx.conf.default
+%config(noreplace) %{_sysconfdir}/nginx/naxsi_core.rules
+%config(noreplace) %{_sysconfdir}/nginx/prometheus.lua
+%config(noreplace) %{_sysconfdir}/nginx/prometheus_test.lua
 %config(noreplace) %{_sysconfdir}/nginx/scgi_params
 %config(noreplace) %{_sysconfdir}/nginx/scgi_params.default
 %config(noreplace) %{_sysconfdir}/nginx/uwsgi_params
